src/search/containers/CurrentSearchPage.js

```diff
@@ -7,19 +8,33 @@ export function mapStateToProps(state) {
   return state.search;
 }
 
-export function mapDispatchToProps(dispatch) {
-  return {
-    handleSearch: (query, page = 1) => {
-      dispatch(searchStart(query, page));
-      return search({ page, query })
-        .then((response) => dispatch(searchLoad({ page, query, ...response })));
-    },
-  };
+function performSearch({dispatch, page, query}) {
+  dispatch(searchStart(query, page));
+  return search({ page, query })
+    .then((response) => dispatch(searchLoad({ page, query, ...response })));
 }
 
-const CurrentSearchPage = connect(
-  mapStateToProps,
-  mapDispatchToProps,
-)(SearchPage);
+export function isLoaded({page, query, state}) {
+  return state.query === query && state.page === page && !state.loading;
+}
+
+export function parsePage(page) {
+  const parsed = parseInt(page, 10);
+  return Number.isNaN(parsed) || parsed < 1 ? 1 : parsed;
```

Comment:
it looks like the `parsed < 1` ternary will not ever be operating on a number. Is that intended?